api/src/opentrons/protocol_engine/commands/command_mapper.py

```diff
@@ -0,0 +1,31 @@
+"""Map command request and model types to other values."""
+from datetime import datetime
+from typing import Any
+
+from .command import CommandStatus
+from .command_unions import Command, CommandRequest
+
+
+class CommandMapper:
+    """Static methods to map and modify command models."""
+
+    @staticmethod
+    def map_request_to_command(
+        request: CommandRequest,
+        command_id: str,
+        created_at: datetime,
+    ) -> Command:
+        """Map a CommandRequest instance to a full command."""
+        # NOTE(mc, 2021-06-22): mypy has trouble with this automatic
+        # request > command routing, but behavior is covered by unit tests
```

Comment:
Obnoxious nitpick, sorry:

```suggestion
        # mypy has trouble with this automatic
        # request > command mapping, but behavior is covered by unit tests
```

* For comments like this, where they primarily explain or justify the code, I tend to omit the `NOTE(...)` signature. The author and timestamp aren't as important as they are in `TODO`s. For example, it's not like there are architectural details in your head about how to avoid this particular thing, and it's not like there's a specific MyPy update at the time of writing that we're waiting for to fix this. (I assume.)
* `routing` -> `mapping` to consistently use one verb for what this method does.

Reply:
I think my intention here was to be more of a `TODO`; it feels conceivable that we could figure out some way to precisely type this (or something about a future mypy update makes it possible). I'll shift it to a TODO and make the `s/routing/mapping/` change